src/runtime/pymodule.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+
+namespace Python.Runtime
+{
+    public class PyModule : PyScope
+    {
+        internal PyModule(ref NewReference reference) : base(ref reference, PyScopeManager.Global) { }
+
+        /// <summary>
+        /// Given a module or package name, import the
+        /// module and return the resulting module object as a <see cref="PyModule"/>.
+        /// </summary>
+        /// <param name="name">Fully-qualified module or package name</param>
+        public static PyModule Import(string name)
+        {
+            NewReference op = Runtime.PyImport_ImportModule(name);
+            PythonException.ThrowIfIsNull(op);
+            return new PyModule(ref op);
+        }
+
+        /// <summary>
+        /// Reloads the module, and returns the updated object
+        /// </summary>
+        public PyModule Reload()
+        {
+            NewReference op = Runtime.PyImport_ReloadModule(this.Reference);
+            PythonException.ThrowIfIsNull(op);
+            return new PyModule(ref op);
+        }
+
+        public static PyModule FromString(string name, string code)
+        {
+            using NewReference c = Runtime.Py_CompileString(code, "none", (int)RunFlagType.File);
```

Comment:
We can do a little better using caller information: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/caller-information

Reply:
If you are suggesting this for `filename` parameter, then passing C# code file path here does not seem like a good idea. Line numbers will be off, and there's no guarantee, that the source code being compiled is present at the caller, rather than somewhere higher in the callstack.

Reply:
It would just allow the user to actually see where the code is coming from. The `filename` of a code object is mostly informational, as far as I can see, and if the respective file is not readily available, it will just not show up in stack traces.

Reply:
I am not going to make this change, as I find it potentially confusing, and of limited usefulness.